src/com/epam/javatrainings/annotation/MinAnnotationProcessor.java

```diff
@@ -0,0 +1,24 @@
+package com.epam.javatrainings.annotation;
+
+import java.lang.reflect.Field;
+import java.time.LocalDate;
+
+public class MinAnnotationProcessor {
+
+    boolean minValidate(CustomerDto instance) {
+        try {
+            Field discount = instance.getClass().getDeclaredField("discountRate");
```

Comment:
Make your annotation GENERIC. For now,  it depends on FIELD NAME. 
Use [this ](https://web-engineering.info/tech/JavaJpaJsf/book/ch06s07.html)link to solve this issue.
The same for the other annotations.